zebra-chain/src/transparent.rs

```diff
@@ -103,6 +105,40 @@ pub enum Input {
     },
 }
 
+impl Input {
+    /// Get the value balance of this input.
+    pub fn value_balance(
+        &self,
+        ordered_utxos: &HashMap<OutPoint, utxo::OrderedUtxo>,
+    ) -> Result<Amount<NegativeAllowed>, InputError> {
+        match self {
+            Input::PrevOut { outpoint, .. } => {
+                let utxos = utxos_from_ordered_utxos(ordered_utxos.clone());
+                if utxos.contains_key(outpoint) {
+                    Ok(utxos[outpoint]
+                        .output
+                        .value
+                        .constrain()
+                        .expect("conversion from NonNegative to NegativeAllowed is always valid"))
+                } else {
+                    Err(InputError::NotEnoughUtxos)
+                }
+            }
+            Input::Coinbase { .. } => Err(InputError::NoCoinBaseAllowed),
```

Comment:
Why do we return an error here? Is there a value we can use instead?

Do coinbase inputs change the value in any pool?